File: app/src/main/java/com/dicoding/projectcapstone/ui/adapter/LokasiAdapter.java
package com.dicoding.projectcapstone.ui.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RatingBar;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.dicoding.projectcapstone.R;
import java.util.ArrayList;
import com.dicoding.projectcapstone.model.Lokasi;


public class LokasiAdapter extends RecyclerView.Adapter<LokasiAdapter.LokasiViewHolder> {

    private ArrayList<Lokasi> lokasiList;

    public LokasiAdapter(ArrayList<Lokasi> lokasiList) {
        this.lokasiList = lokasiList;
    }

    @NonNull
    @Override
    public LokasiViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_lokasi, parent, false);
        return new LokasiViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull LokasiViewHolder holder, int position) {
        Lokasi lokasi = lokasiList.get(position);
        holder.nameTextView.setText(lokasi.getName());
        holder.distanceTextView.setText(lokasi.getDistance());
        holder.timeTextView.setText(lokasi.getTime());
        holder.imageView.setImageResource(lokasi.getImageResource());
        holder.ratingBar.setRating((float) lokasi.getRating());
    }

    @Override
    public int getItemCount() {
        return lokasiList.size();
    }

    public static class LokasiViewHolder extends RecyclerView.ViewHolder {

        TextView nameTextView, distanceTextView, timeTextView;
        ImageView imageView;
        RatingBar ratingBar;

        public LokasiViewHolder(@NonNull View itemView) {
            super(itemView);
            nameTextView = itemView.findViewById(R.id.text_name);
            distanceTextView = itemView.findViewById(R.id.text_distance);
            timeTextView = itemView.findViewById(R.id.text_time);
            imageView = itemView.findViewById(R.id.image_location);
            ratingBar = itemView.findViewById(R.id.rating_location);
        }
    }
}

File: app/src/main/java/com/dicoding/projectcapstone/location/adapter/LokasiAdapter.kt
package com.dicoding.projectcapstone.location.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RatingBar
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.dicoding.projectcapstone.R
import com.dicoding.projectcapstone.api.RetrofitClient
import com.dicoding.projectcapstone.location.model.Lokasi
import com.dicoding.projectcapstone.product.DataItem

class LokasiAdapter(private val lokasiList: ArrayList<Lokasi>, private val onItemClick: (Lokasi) -> Unit
) : RecyclerView.Adapter<LokasiAdapter.LokasiViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LokasiViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_lokasi, parent, false)
        return LokasiViewHolder(view)
    }

    override fun onBindViewHolder(holder: LokasiViewHolder, position: Int) {
        val lokasi = lokasiList[position]
        holder.nameTextView.text = lokasi.name
        Glide.with(holder.itemView.context)
            .load(lokasi.imageUrl)
            .apply(
                RequestOptions()
                    .placeholder(R.drawable.ic_launcher_background)
                    .error(R.drawable.ic_launcher_background)
            )
            .into(holder.imageView)
        holder.ratingBar.rating = lokasi.rating.toFloat()
        holder.ratingText.text = lokasi.rating.toString()
        holder.distanceView.text = String.format("%.2f km", lokasi.distance)
        if (lokasi.isOpen) {
            holder.itemStatus.text = "Buka"
            holder.itemStatus.setTextColor(0xFF4CAF50.toInt())
        } else {
            holder.itemStatus.text = "Tutup"
            holder.itemStatus.setTextColor(0xFFF44336.toInt())
        }
        holder.itemView.setOnClickListener {
            onItemClick(lokasi)
        }
    }

    override fun getItemCount(): Int {
        return lokasiList.size
    }

    class LokasiViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var nameTextView: TextView =
            itemView.findViewById(R.id.text_name)
        var imageView: ImageView =
            itemView.findViewById(R.id.image_location)
        var distanceView: TextView =
            itemView.findViewById(R.id.item_distance)
        var ratingBar: RatingBar = itemView.findViewById(R.id.rating_location)
        var ratingText: TextView =
            itemView.findViewById(R.id.text_rating_value)
        var itemStatus: TextView =
            itemView.findViewById(R.id.item_status)
    }

    fun addLokasi(lokasi: Lokasi) {
        lokasiList.add(lokasi)
    }
}